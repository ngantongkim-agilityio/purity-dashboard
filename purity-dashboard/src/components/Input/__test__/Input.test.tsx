import { render, fireEvent } from '@testing-library/react';

// Components
import { Input } from '..';

describe('Input component', () => {
  it('Should render Input component with default props', () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  it('Simulate onChange event when fill value', () => {
    const handleChange = jest.fn();

    const { getByTestId } = render(
      <Input data-testid='test-input' onChange={handleChange} />
    );
    const inputElement = getByTestId('test-input');
    fireEvent.change(inputElement, { target: { value: 'test' } });

    expect(handleChange).toHaveBeenCalled();
    expect(inputElement).toHaveValue('test');
  });

  test('Should display error message when value is invalid', () => {
    const { getByText } = render(
      <Input placeholder='Password' isInvalid errorMessage='Invalid password' />
    );
    const errorMessage = getByText('Invalid password');

    expect(errorMessage).toBeInTheDocument();
  });
});
