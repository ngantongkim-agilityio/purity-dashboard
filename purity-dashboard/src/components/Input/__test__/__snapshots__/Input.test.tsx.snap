// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render Input component with default props 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
    data-has-value="true"
    data-slot="base"
  >
    <div
      class="flex flex-col h-20"
      data-slot="main-wrapper"
    >
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 border-medium data-[hover=true]:border-default-400 group-data-[focus=true]:border-default-foreground min-h-10 transition-background !duration-150 transition-colors motion-reduce:transition-none bg-primary-100 border-secondary-50 data-[hover=true]:!bg-primary-100 group-data-[focus=true]:!bg-primary-100 data-[focus=true]:!border-secondary-200 group-data-[invalid=true]:border-danger h-auto py-4 rounded-2xl"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full items-center h-full box-border"
          data-slot="inner-wrapper"
        >
          <input
            aria-label=" "
            class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-secondary-300 placeholder:text-secondary-200 bg-primary-100 text-sm mx-2 group-data-[invalid=true]:!text-danger rounded"
            data-slot="input"
            id="react-aria-:r0:"
            title=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
